tests/integration/telemetry/stackdriver/stackdriver_filter_test.go

```diff
@@ -155,22 +163,38 @@ func setupConfig(_ resource.Context, cfg *istio.Config) {
 	cfg.ControlPlaneValues = `
 meshConfig:
   enableTracing: true
-values:
-  telemetry:
-    v2:
-      stackdriver:
-        configOverride:
-          meshEdgesReportingDuration: "5s"
-          enable_mesh_edges_reporting: true
 `
 	// enable stackdriver filter
 	cfg.Values["telemetry.v2.stackdriver.enabled"] = "true"
 	cfg.Values["telemetry.v2.stackdriver.logging"] = "true"
 	cfg.Values["telemetry.v2.stackdriver.topology"] = "true"
 	cfg.Values["telemetry.v2.stackdriver.configOverride.enable_audit_log"] = "true"
+	cfg.Values["telemetry.v2.stackdriver.configOverride.meshEdgesReportingDuration"] = "5s"
+	cfg.Values["telemetry.v2.stackdriver.configOverride.enable_mesh_edges_reporting"] = "true"
 	cfg.Values["global.proxy.tracer"] = "stackdriver"
 	cfg.Values["pilot.traceSampling"] = "100"
 	cfg.Values["telemetry.v2.accessLogPolicy.enabled"] = "true"
+
+	// conditionally use a fake metadata server for testing off of GCP
+	if gceInst != nil {
+		cfg.MetadataServer = gceInst.Address()
+		cfg.ControlPlaneValues = strings.Join(
+			[]string{
+				cfg.ControlPlaneValues,
+				fakeGCEMetadataServerDefaultProxyConfig,
+				gceInst.Address()},
+			"")
+	}
+}
+
+// if not running on GCP, we want to set up a fake GCP Metadata server for
+// purposes of this test
+func conditionallySetupMetadataServer(ctx resource.Context) (err error) {
```

Comment:
nit: should it be `conditionallySetupFakeMetadataServer`?

Reply:
done.